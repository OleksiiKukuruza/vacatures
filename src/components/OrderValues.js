import React from 'react';
import PropTypes from 'prop-types';
import { Row, Cell } from './Table';
import styled from 'styled-components';

export const StyledOrderValues = styled.div`
  border-top: 1px solid ${props => props.theme.primary.contrastText};
  border-bottom: 1px solid ${props => props.theme.primary.contrastText};
  padding: ${props => 3 * props.theme.spacingUnit}px 0;
`;

export const StyledRow = styled(Row)`
  padding: ${props => props.theme.spacingUnit / 2}px 0;
`;

const StyledBoldCell = styled(Cell)`
  font-weight: bold;
`;

const OrderValues = ({ name, originalPrice, discount }) => (
  <StyledOrderValues>
    <StyledRow>
      <Cell size={10}>
        Prijs {name}
      </Cell>
      <Cell size={2} justifyContent="flex-end">
        € {originalPrice}
      </Cell>
    </StyledRow>
    <StyledRow>
      <Cell size={10}>
        Korting
      </Cell>
      <Cell size={2} justifyContent="flex-end">
        - € {discount}
      </Cell>
    </StyledRow>
    <StyledRow>
      <StyledBoldCell size={10}>
        Totaal
      </StyledBoldCell>
      <StyledBoldCell size={2} justifyContent="flex-end">
        € {originalPrice - discount}
      </StyledBoldCell>
    </StyledRow>
  </StyledOrderValues>
);

OrderValues.propTypes = {
  name: PropTypes.string.isRequired,
  originalPrice: PropTypes.number.isRequired,
  discount: PropTypes.number.isRequired
};

export default OrderValues;
