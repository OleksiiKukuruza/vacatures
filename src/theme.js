const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  spacingUnit: 8
};

export default theme;
