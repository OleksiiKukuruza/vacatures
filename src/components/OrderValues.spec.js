import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import OrderValues, { StyledOrderValues, StyledRow } from './OrderValues';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('OrderValues', () => {
  it('renders OrderValues without crashing', () => {
    const props = { name: 'name', originalPrice: 100, discount: 10 };
    const { wrapper } = setup(OrderValues, props);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledOrderValues without crashing', () => {
    const { wrapper } = setup(StyledOrderValues, { theme });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledRow without crashing', () => {
    const { wrapper } = setup(StyledRow, { theme });
    expect(wrapper).toMatchSnapshot();
  });
});
