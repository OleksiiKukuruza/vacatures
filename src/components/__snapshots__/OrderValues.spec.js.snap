// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderValues renders OrderValues without crashing 1`] = `
<styled.div>
  <Styled(styled.div)>
    <styled.div
      size={10}
    >
      Prijs 
      name
    </styled.div>
    <styled.div
      justifyContent="flex-end"
      size={2}
    >
      € 
      100
    </styled.div>
  </Styled(styled.div)>
  <Styled(styled.div)>
    <styled.div
      size={10}
    >
      Korting
    </styled.div>
    <styled.div
      justifyContent="flex-end"
      size={2}
    >
      - € 
      10
    </styled.div>
  </Styled(styled.div)>
  <Styled(styled.div)>
    <Styled(styled.div)
      size={10}
    >
      Totaal
    </Styled(styled.div)>
    <Styled(styled.div)
      justifyContent="flex-end"
      size={2}
    >
      € 
      90
    </Styled(styled.div)>
  </Styled(styled.div)>
</styled.div>
`;

exports[`OrderValues renders StyledOrderValues without crashing 1`] = `
.c0 {
  border-top: 1px solid #fff;
  border-bottom: 1px solid #fff;
  padding: 24px 0;
}

<div
  className="c0"
/>
`;

exports[`OrderValues renders StyledRow without crashing 1`] = `
.c0 {
  padding: 4px 0;
}

<styled.div
  className="c0"
  theme={
    Object {
      "background": "#f1f4f5",
      "error": "#f44336",
      "primary": Object {
        "contrastText": "#fff",
        "main": "#54b6c8",
      },
      "spacingUnit": 8,
    }
  }
/>
`;
